nix/mobile/android/targets/release-android.nix

```diff
@@ -0,0 +1,129 @@
+{ stdenv, stdenvNoCC, lib, target-os, callPackage,
+  mkFilter, bash, file, gnumake, watchman, gradle, androidEnvShellHook, mavenAndNpmDeps, nodejs, openjdk, prod-build, status-go, zlib }:
+
+{ build-number ? "9999",
+  build-type ? "nightly", # Build type (e.g. nightly, release, e2e). Default is to use .env.nightly file
+  gradle-opts ? "",
+  keystore-file ? "", # Path to the .keystore file used to sign the package
+  secrets-file ? "", # Path to the file containing secret environment variables
+  env ? {} # Attribute set contaning environment variables to expose to the build script
+}:
+
+let
+  name = "release-${target-os}";
+  gradleHome = "$NIX_BUILD_TOP/.gradle";
+  localMavenRepo = "${mavenAndNpmDeps.buildInputs}/.m2/repository";
+  sourceProjectDir = "${mavenAndNpmDeps.buildInputs}/project";
+  envFileName =
+    if (build-type == "release" || build-type == "nightly" || build-type == "e2e") then ".env.${build-type}" else
+    if build-type != "" then ".env.jenkins" else ".env";
+  buildType' = if (build-type == "release" || build-type == "nightly") then "release" else "pr";
+  generatedApkPath = "android/app/build/outputs/apk/${buildType'}/app-${buildType'}.apk";
+  outApkName = "app.apk";
+
+in stdenv.mkDerivation {
+  inherit name;
+  src =
+    let path = ./../../../..;
+    in builtins.path { # We use builtins.path so that we can name the resulting derivation, otherwise the name would be taken from the checkout directory, which is outside of our control
+      inherit path;
+      name = "status-react-${name}-source";
+      filter =
+        # Keep this filter as restrictive as possible in order to avoid unnecessary rebuilds and limit closure size
+        mkFilter {
+          dirRootsToInclude = [ 
+            "mobile_files"
+            "modules/react-native-status"
+            "packager"
+            "resources"
+          ];
+          dirsToExclude = [ ".git" ".svn" "CVS" ".hg" ".gradle" "build" "intermediates" "libs" "obj" ];
+          filesToInclude = [ envFileName "STATUS_GO_VERSION" "VERSION" ];
+          root = path;
+        };
+    };
+  nativeBuildInputs = [ bash gradle ] ++ lib.optionals stdenv.isDarwin [ file gnumake watchman ];
+  buildInputs = [ nodejs openjdk ];
+  phases = [ "unpackPhase" "patchPhase" "buildPhase" "installPhase" ];
+  postUnpack = ''
+    mkdir -p ${gradleHome}
+
+    ${if keystore-file != "" then "cp -a --no-preserve=ownership ${keystore-file} ${gradleHome}/; export STATUS_RELEASE_STORE_FILE=${gradleHome}/$(basename ${keystore-file})" else ""}
+
+    # Ensure we have the right .env file
+    cp -f $sourceRoot/${envFileName} $sourceRoot/.env
+
+    # Copy index.*.js input file
+    cp -a --no-preserve=ownership ${prod-build}/index*.js $sourceRoot/
+
+    # Copy android/ directory
```

Comment:
I thought `builtins.path` and `mkFIlter` was already taking care of this stuff.

Reply:
Well here we're copying from the prod-build-android step (or something like that, i'm on mobile now), not from the user repo.